include/drivers/pinctrl/pinctrl_soc_mcux_rt_common.h

```diff
@@ -0,0 +1,93 @@
+	uint32_t input_register;
+	uint32_t input_daisy;
+	uint32_t config_register;
+};
+
+typedef struct pinctrl_soc_pin {
+	struct pinctrl_soc_pinmux pinmux;
+	uint32_t pin_ctrl_flags;
+} pinctrl_soc_pin_t;
+
+#define MCUX_RT_INPUT_SCHMITT_ENABLE_SHIFT 0
+#define MCUX_RT_INPUT_SCHMITT_ENABLE(x) ((x >> MCUX_RT_INPUT_SCHMITT_ENABLE_SHIFT) & 0x1)
+#define MCUX_RT_DRIVE_OPEN_DRAIN_SHIFT 1
+#define MCUX_RT_DRIVE_OPEN_DRAIN(x) ((x >> MCUX_RT_DRIVE_OPEN_DRAIN_SHIFT) & 0x1)
+#define MCUX_RT_INPUT_ENABLE_SHIFT 2
+#define MCUX_RT_INPUT_ENABLE(x) ((x >> MCUX_RT_INPUT_ENABLE_SHIFT) & 0x1)
+#define MCUX_RT_BIAS_BUS_HOLD_SHIFT 3
+#define MCUX_RT_BIAS_BUS_HOLD(x) ((x >> MCUX_RT_BIAS_BUS_HOLD_SHIFT) & 0x1)
+#define MCUX_RT_BIAS_PULL_DOWN_SHIFT 4
+#define MCUX_RT_BIAS_PULL_DOWN(x) ((x >> MCUX_RT_BIAS_PULL_DOWN_SHIFT) & 0x1)
+#define MCUX_RT_BIAS_PULL_UP_SHIFT 5
+#define MCUX_RT_BIAS_PULL_UP(x) ((x >> MCUX_RT_BIAS_PULL_UP_SHIFT) & 0x3)
+#define MCUX_RT_DRIVE_STRENGTH_SHIFT 7
+#define MCUX_RT_DRIVE_STRENGTH(x) ((x >> MCUX_RT_DRIVE_STRENGTH_SHIFT) & 0x7)
+#define MCUX_RT_SPEED_SHIFT 10
+#define MCUX_RT_SPEED(x) ((x >> MCUX_RT_SPEED_SHIFT) & 0x3)
+#define MCUX_RT_SLEW_RATE_SHIFT 12
+#define MCUX_RT_SLEW_RATE(x) ((x >> MCUX_RT_SLEW_RATE_SHIFT) & 0x1)
+
+
+#define Z_PINCTRL_MCUX_RT_PINCFG_INIT(node_id)				\
+	 ((DT_PROP(node_id, input_schmitt_enable) << MCUX_RT_INPUT_SCHMITT_ENABLE_SHIFT) |	\
+	 (DT_PROP(node_id, drive_open_drain) << MCUX_RT_DRIVE_OPEN_DRAIN_SHIFT) |	\
+	 (DT_PROP(node_id, input_enable) << MCUX_RT_INPUT_ENABLE_SHIFT) |	\
+	 (DT_PROP(node_id, bias_bus_hold) << MCUX_RT_BIAS_BUS_HOLD_SHIFT) |	\
+	 (DT_PROP(node_id, bias_pull_down) << MCUX_RT_BIAS_PULL_DOWN_SHIFT) |	\
+	 (DT_ENUM_IDX(node_id, drive_strength) << MCUX_RT_DRIVE_STRENGTH_SHIFT) |\
+	 (DT_ENUM_IDX(node_id, bias_pull_up) << MCUX_RT_BIAS_PULL_UP_SHIFT) |	\
+	 (DT_ENUM_IDX(node_id, slew_rate) << MCUX_RT_SLEW_RATE_SHIFT) |	\
+	 (DT_ENUM_IDX(node_id, nxp_speed) << MCUX_RT_SPEED_SHIFT))
+
+#define Z_PINCTRL_PIN_INIT(node_id)				\
+	{ .pinmux.mux_register = DT_PROP_BY_IDX(node_id, pinmux, 0),	\
+	  .pinmux.mux_mode = DT_PROP_BY_IDX(node_id, pinmux, 1),	\
+	  .pinmux.input_register = DT_PROP_BY_IDX(node_id, pinmux, 2),	\
+	  .pinmux.input_daisy = DT_PROP_BY_IDX(node_id, pinmux, 3),	\
+	  .pinmux.config_register = DT_PROP_BY_IDX(node_id, pinmux, 4),	\
+	  .pin_ctrl_flags = Z_PINCTRL_MCUX_RT_PINCFG_INIT(node_id),	\
+	},
+
+#define Z_PINCTRL_STATE_PIN_INIT(node_id, prop, idx)		\
+	Z_PINCTRL_PIN_INIT(DT_PROP_BY_IDX(node_id, prop, idx))
+
+#define Z_PINCTRL_STATE_PINS_INIT(node_id, prop)			\
+	{DT_FOREACH_PROP_ELEM(node_id, prop, Z_PINCTRL_STATE_PIN_INIT)}
+
+/**
+ * @brief Define all the states for the given node identifier.
+ *
+ * @param node_id Node identifier.
+ */
+#define Z_PINCTRL_STATES_DEFINE(node_id)				\
+	static const struct pinctrl_state				\
+	Z_PINCTRL_STATES_NAME(node_id)[] = {				\
+		UTIL_LISTIFY(DT_NUM_PINCTRL_STATES(node_id),		\
+			     Z_PINCTRL_STATE_INIT, node_id)		\
+	};
```

Comment:
`Z_PINCTRL_STATES_DEFINE` is already defined in `./include/drivers/pinctrl.h`

Reply:
Thanks for the heads up, removed in latest push